common/metric_types.py

```
@@ -288,6 +298,38 @@ async def _send_request(
             json=self._base_request,
         )
 
+    def _on_json_response(self, json_response: dict[str, Any]) -> None:
+        """Hook called after a successful JSON-RPC response.
+
+        Override in subclasses to capture response fields (e.g. block number).
+        The default implementation does nothing.
+        """
+        pass
```

Ruff (0.15.9)

[warning] Unnecessary `pass` statement

Remove unnecessary `pass`

(PIE790)